lib/zipkin-tracer/faraday/zipkin-tracer.rb

```diff
@@ -20,32 +20,40 @@ def initialize(app, service_name = nil)
     end
 
     def call(env)
-      # handle either a URI object (passed by Faraday v0.8.x in testing), or something string-izable
-      url = env[:url].respond_to?(:host) ? env[:url] : URI.parse(env[:url].to_s)
-      local_endpoint = Trace.default_endpoint # The rack middleware set this up for us.
-      remote_endpoint = callee_endpoint(url, local_endpoint.ip_format) # The endpoint we are calling.
-      response = nil
       trace_id = Trace.id.next_id
       with_trace_id(trace_id) do
         B3_HEADERS.each do |method, header|
           env[:request_headers][header] = trace_id.send(method).to_s
         end
-        # annotate with method (GET/POST/etc.) and uri path
-        @tracer.set_rpc_name(trace_id, env[:method].to_s.downcase)
-        @tracer.record(trace_id, Trace::BinaryAnnotation.new('http.uri', url.path, 'STRING', local_endpoint))
-        @tracer.record(trace_id, Trace::BinaryAnnotation.new('sa', '1', 'BOOL', remote_endpoint))
-        @tracer.record(trace_id, Trace::Annotation.new(Trace::Annotation::CLIENT_SEND, local_endpoint))
-        response = @app.call(env).on_complete do |renv|
-          # record HTTP status code on response
-          @tracer.record(trace_id, Trace::BinaryAnnotation.new('http.status', renv[:status].to_s, 'STRING', local_endpoint))
+        if trace_id.sampled?
+          trace!(env, trace_id)
+        else
+          @app.call(env)
         end
-        @tracer.record(trace_id, Trace::Annotation.new(Trace::Annotation::CLIENT_RECV, local_endpoint))
       end
-      response
     end
 
     private
 
+    def trace!(env, trace_id)
+      response = nil
+      # handle either a URI object (passed by Faraday v0.8.x in testing), or something string-izable
+      url = env[:url].respond_to?(:host) ? env[:url] : URI.parse(env[:url].to_s)
+      local_endpoint = Trace.default_endpoint # The rack middleware set this up for us.
+      remote_endpoint = callee_endpoint(url, local_endpoint.ip_format) # The endpoint we are calling.
+      # annotate with method (GET/POST/etc.) and uri path
+      @tracer.set_rpc_name(trace_id, env[:method].to_s.downcase)
+      @tracer.record(trace_id, Trace::BinaryAnnotation.new('http.uri', url.path, 'STRING', local_endpoint))
```

Comment:
Applies for the `http.uri` and other `sa` too. 
Especially in the `sa` case it would allow us to have an easier to understand constant name than this acronym.